### PR TITLE
PERA-1374 :: Migrate account creation flow

### PR DESCRIPTION
**Pull request by mitsinsar**
# Pull Request Template

## Description
- Migrated account creation flow except the part requires ledger connection. This will be implemented after txn signing refactor

## Related Issues
- Closes https://algorandfoundation.atlassian.net/browse/PERA-1374

## Checklist
- [X] Have you tested your changes locally?
- [X] Have you reviewed the code for any potential issues?
- [  ] Have you documented any necessary changes in the project's documentation?
- [X] Have you added any necessary tests for your changes?
- [X] Have you updated any relevant dependencies?

### CODE REVIEW

**Review by michaeltchuang — APPROVED** [review]
i think this is fine to merge, we can review again when we need to merge into dev

File: composeTestApp/src/commonMain/kotlin/co/algorand/app/ui/screens/accounts/AccountsViewModel.kt [hunk 1]
```diff
@@ -15,11 +15,10 @@ package co.algorand.app.ui.screens.accounts
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import co.algorand.app.ui.screens.accounts.AccountsViewModel.ViewState
+import com.algorand.common.account.core.domain.usecase.AddAlgo25Account
+import com.algorand.common.account.core.domain.usecase.AddBip39Account
```
Comment (ehanoc) [inline composeTestApp/src/commonMain/kotlin/co/algorand/app/ui/screens/accounts/AccountsViewModel.kt:19]: @michaeltchuang as we spoke before; can we **NOT** call accounts "BIP39" accounts. This a smell to me that we are not understanding what the BIP39 / BIP32 / BIP44 stand for . And it's those 3 together that gives wallets the ability to do HD (Hierarchical derivation)

I suggest we read the BIP39 / BIP32 / BIP44 specifications. We can name some more appropriate then? 

CC: @mitsinsar

**Comment by ehanoc** [comment]
@mitsinsar @michaeltchuang Where can i find reports for test coverage? % branch, % functions, % lines, etc

File: composeTestApp/src/commonMain/kotlin/co/algorand/app/ui/screens/accounts/AccountsViewModel.kt (hunk 1 above)
Comment (michaeltchuang) [reply composeTestApp/src/commonMain/kotlin/co/algorand/app/ui/screens/accounts/AccountsViewModel.kt:19]: This isn't going into `dev` branch yet, there will be follow up PRs into the parent feature branch to rename the HD wallets portions

**Pull request merged**